### PR TITLE
edhoc-hacspec: declassify numbers before return from parse

### PR DESCRIPTION
**Pull request by geonnave**
Addressing #49.

### CODE REVIEW

File: hacspec/out/Edhoc_hacspec.fst [hunk 1]
```diff
@@ -0,0 +1,1129 @@
+module Edhoc_hacspec
```
Comment (malishav) [inline hacspec/out/Edhoc_hacspec.fst:1]: Why is the .fst file part of this PR?

File: hacspec/src/lib.rs [hunk 2]
```diff
@@ -60,9 +60,9 @@ pub fn r_process_message_1(
         let (method, supported_suites, g_x, c_i) = parse_message_1(message_1);
 
         // verify that the method is supported
-        if method.declassify() == EDHOC_METHOD {
+        if method == EDHOC_METHOD {
             // Step 2: verify that the selected cipher suite is supported
-            if supported_suites[0u8].declassify() == EDHOC_SUPPORTED_SUITES[0u8].declassify() {
+            if supported_suites[0] == EDHOC_SUPPORTED_SUITES[0u8].declassify() {
```
Comment (malishav) [inline hacspec/src/lib.rs:65]: Let's homogenize this so `declassify` does not appear any more in `*_process_message`?

File: hacspec/src/lib.rs [hunk 3]
```diff
@@ -220,7 +220,7 @@ pub fn r_process_message_3(
             let (kid, mac_3) = decode_plaintext_3(&plaintext_3);
 
             // compare the kid received with the kid expected in id_cred_i
-            if kid.declassify() == id_cred_i_expected[id_cred_i_expected.len() - 1].declassify() {
+            if kid == id_cred_i_expected[id_cred_i_expected.len() - 1].declassify() {
```
Comment (malishav) [inline hacspec/src/lib.rs:223]: same comment as above

File: hacspec/src/lib.rs [hunk 4]
```diff
@@ -420,7 +420,7 @@ pub fn i_process_message_2(
 
         // Check MAC before checking KID
         if mac_2.declassify_eq(&expected_mac_2) {
-            if kid.declassify() == id_cred_r_expected[id_cred_r_expected.len() - 1].declassify() {
+            if kid == id_cred_r_expected[id_cred_r_expected.len() - 1].declassify() {
```
Comment (malishav) [inline hacspec/src/lib.rs:423]: see above

**Review by malishav — COMMENTED** [review]
Thanks for hte PR! My main comment/question is whether we can homogenize the code such that `declassify` does not appear in top-level functions? With the current code, some parts are declassified and some are not, which is not ideal IMO

File: hacspec/out/Edhoc_hacspec.fst (hunk 1 above)
Comment (geonnave) [reply hacspec/out/Edhoc_hacspec.fst:1]: Committed it by mistake... removing.

**Comment by geonnave** [comment]
> whether we can homogenize the code such that `declassify` does not appear in top-level functions? With the current code, some parts are declassified and some are not, which is not ideal IMO

I think it is a good idea to be more homogeneous. Just checking how we would do it:
- update `EDHOC_SUPPORTED_SUITES` to not use secret integers (not sure if this makes sense)
- modify `id_cred_[ir]_expected` to `[u8; ID_CRED_LEN]`, in `* _process_message_*` (but then I would have to call `from_public_slice` before passing it to functions such as `compute_mac_3`.

**Comment by malishav** [comment]
> update EDHOC_SUPPORTED_SUITES to not use secret integers (not sure if this makes sense)

Why not? Is it used elsewhere? It seems the cleanest IMO

File: hacspec/src/lib.rs [hunk 5]
```diff
@@ -899,7 +900,10 @@ fn encode_kdf_context(
     // encode context in line
     // assumes ID_CRED_R and CRED_R are already CBOR-encoded
     let mut output = BytesMaxContextBuffer::new();
-    output = output.update(0, id_cred);
+    for i in 0..id_cred.len() {
```
[the remark this answers is not in the document]
Comment (malishav) [reply hacspec/src/lib.rs:903]: Oh, I see why.. `output` is a hacspec type.. nevermind the comment

[the remark this answers is not in the document]
Comment (geonnave) [reply hacspec/src/lib.rs:903]: Done.

File: hacspec/src/lib.rs [hunk 6]
```diff
@@ -1292,7 +1297,7 @@ mod tests {
     fn test_compute_mac_3() {
         let prk_4e3m_tv = BytesHashLen::from_hex(PRK_4E3M_TV);
         let th_3_tv = BytesHashLen::from_hex(TH_3_TV);
-        let id_cred_i_tv = BytesIdCred::from_hex(ID_CRED_I_TV);
```
[the remark this answers is not in the document]
Comment (geonnave) [reply hacspec/src/lib.rs:1295]: Done.

[the remark this answers is not in the document]
Comment (malishav) [reply hacspec/src/lib.rs:1295]: Could you also update it ssimilarly for other occurences, noted with "same comment"

[the remark this answers is not in the document]
Comment (geonnave) [reply hacspec/src/lib.rs:1295]: Oh sure, done.

**Comment by geonnave** [comment]
Still needed: test fstar generation after type changes.
Obstacle: `cargo circus` not working in my personal computer.
Solution: wait for a fix from the `hacspec` team, or go to Inria run the tests (maybe tomorrow).

**Pull request closed without merging**